crates/runner/src/memory/mod.rs

```diff
@@ -14,8 +14,9 @@ const M31S_IN_QM31: usize = 4;
 
 /// The maximum number of bits for a memory address, set to 30.
 /// This limits the memory size to 2^30 elements.
-// TODO: check with Starkware
-const MAX_MEMORY_SIZE_BITS: u8 = 30;
+const MAX_MEMORY_SIZE_BITS: u8 = 28;
+
+const MAX_ADDRESS: usize = (1 << MAX_MEMORY_SIZE_BITS) - 1;
```

Comment:
the max address defined in validate_address is not the same
`let max_addr = 1 << MAX_MEMORY_SIZE_BITS;`